File: app/src/main/java/com/example/recipeclipper/ui/navigation/ComingSoonScreen.kt
package com.example.recipeclipper.ui.navigation

import androidx.annotation.StringRes
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.res.stringResource
import androidx.compose.ui.unit.dp
import com.example.recipeclipper.R
import com.example.recipeclipper.ui.recipe.Hairline
import com.example.recipeclipper.ui.theme.RecipeClipperTheme

/**
 * The Week, Groceries and Pantry tabs until their features land (#49–#51): the tab's name, one
 * line on what it will hold, and "Coming soon". Deliberately quiet, with nothing to tap.
 */
@Composable
fun ComingSoonScreen(@StringRes title: Int, @StringRes description: Int) {
    RecipeClipperTheme {
        Surface(Modifier.fillMaxSize(), color = MaterialTheme.colorScheme.background) {
            Column(Modifier.padding(start = 20.dp, end = 20.dp, top = 24.dp)) {
                Text(stringResource(title), style = MaterialTheme.typography.headlineMedium)
                Spacer(Modifier.height(12.dp))
                Hairline()
                Spacer(Modifier.height(24.dp))
                Text(
                    stringResource(description),
                    style = MaterialTheme.typography.bodyLarge,
                    color = MaterialTheme.colorScheme.onSurfaceVariant
                )
                Spacer(Modifier.height(8.dp))
                Text(
                    stringResource(R.string.coming_soon),
                    style = MaterialTheme.typography.labelMedium,
                    color = MaterialTheme.colorScheme.tertiary
                )
            }
        }
    }
}

File: ios/RecipeClipper/UI/Navigation/ComingSoonScreen.swift
import SwiftUI
import UIKit

/// The Week, Groceries and Pantry tabs until their features land (#49–#51): the tab's name, one
/// line on what it will hold, and "Coming soon". Deliberately quiet, with nothing to tap.
struct ComingSoonScreen: View {
    let title: String
    let description: String

    var body: some View {
        ScrollView {
            VStack(alignment: .leading, spacing: 0) {
                ScreenTitle(title).padding(.bottom, 12)
                Hairline()
                Text(description)
                    .textStyle(Typography.bodyLarge)
                    .foregroundStyle(Palette.muted)
                    .frame(maxWidth: .infinity, alignment: .leading)
                    .padding(.top, 24)
                Text(Strings.comingSoon)
                    .textStyle(Typography.labelMedium)
                    .foregroundStyle(Palette.accentText)
                    .padding(.top, 8)
            }
            .padding(.horizontal, 20)
            .padding(.top, 4)
            .padding(.bottom, 32)
        }
        .screenBackground()
        .navigationBarTitleDisplayMode(.inline)
    }
}

/// The tab bar in the app's tokens: ground with a hairline above it, paprika for the open tab,
/// muted for the rest, Karla labels. UIKit draws the bar, so this goes through its appearance
/// proxy, once, before the first bar is made.
enum TabBarStyle {
    static let apply: Void = {
        let appearance = UITabBarAppearance()
        appearance.configureWithOpaqueBackground()
        appearance.backgroundColor = UIColor(Palette.background)
        appearance.shadowColor = UIColor(Palette.hairline)
        let font = AppFont.karla(11, weight: 500)
        for item in [appearance.stackedLayoutAppearance, appearance.inlineLayoutAppearance,
                     appearance.compactInlineLayoutAppearance] {
            item.normal.iconColor = UIColor(Palette.muted)
            item.normal.titleTextAttributes = [.foregroundColor: UIColor(Palette.muted), .font: font]
            item.selected.iconColor = Palette.uiAccentText
            item.selected.titleTextAttributes = [.foregroundColor: Palette.uiAccentText, .font: font]
        }
        UITabBar.appearance().standardAppearance = appearance
        UITabBar.appearance().scrollEdgeAppearance = appearance
    }()
}
